File: backend/src/models/application.py
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy import Integer, String, DateTime, ForeignKey
from ..database.base import Base


class Application(Base):
    __tablename__ = "applications"
    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id"), nullable=False)
    company: Mapped[str] = mapped_column(String(255), nullable=False) 
    role: Mapped[str] = mapped_column(String(255), nullable=False)
    job_url: Mapped[str] = mapped_column(String(255), nullable=False)
    location: Mapped[str] = mapped_column(String(255), nullable=False)
    status: Mapped[str] = mapped_column(String(255), nullable=False)
    applied_date: Mapped[DateTime] = mapped_column(DateTime, nullable=False)
    follow_up_date: Mapped[DateTime] = mapped_column(DateTime, nullable=True) 
    notes: Mapped[str] = mapped_column(String(255), nullable=True)
    created_at: Mapped[DateTime] = mapped_column(DateTime, nullable=False)    
    
    

File: frontend/src/types/application.ts
export type ApplicationStatus = 'wishlist' | 'applied' | 'interviewing' | 'closed'

export type CompanyTone = 'blue' | 'black' | 'cream' | 'orange' | 'purple' | 'green'

export interface Application {
  id: string
  user_id?: string
  company: string
  role: string
  status: ApplicationStatus | string
  job_url?: string | null
  location?: string | null
  applied_date?: string | null
  follow_up_date?: string | null
  notes?: string | null
  created_at?: string
  tone?: CompanyTone
}

export interface CreateApplicationInput {
  company: string
  role: string
  status: ApplicationStatus
  job_url?: string
  location?: string
  notes?: string
  applied_date?: string
  follow_up_date?: string
}

export interface UpdateApplicationInput {
  company?: string
  role?: string
  status?: ApplicationStatus
  job_url?: string
  location?: string
  notes?: string
  applied_date?: string
  follow_up_date?: string
}

export interface User {
  id: string
  username: string
  email: string
  created_at?: string
}

export interface BoardColumnDef {
  id: ApplicationStatus
  label: string
  dotClass: string
}

export interface SummaryStats {
  total: number
  responseRate: number
  interviewsThisWeek: number
  lastSyncedText: string
}

export interface ToastMessage {
  id: string
  type: 'success' | 'error' | 'info'
  text: string
}
